src/main/java/org/example/service/AnimalService.java

```diff
@@ -0,0 +1,77 @@
+package org.example.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.example.model.Animal;
+
+import java.io.*;
+import java.util.*;
+
+public class AnimalService {
+
+    File file = new File("src/main/resources/animal.json");
+
+    public void serializeList(Animal animal) {
+        JsonMapper jsonMapper = new JsonMapper();
+        try {
+            if (deserializeList() != null) {
+                List<Animal> animals = deserializeList();
+                animals.add(animal);
+                jsonMapper.writeValue(file, animals);
+            } else {
+                jsonMapper.writeValue(file, List.of(animal));
+            }
+        } catch (IOException e) {
+            System.out.println("Can not create file: " + file);
+        }
+    }
+
+    public List<Animal> deserializeList() {
+        JsonMapper jsonMapper = new JsonMapper();
+        try {
+            return jsonMapper.readValue(file, new TypeReference<>() {});
+        } catch (IOException e) {
+            System.out.println("Can not read file: " + file);
+            return null;
```

Comment:
it is better to return an empty list, return a new ArrayList(). The null can case trouble